week-7/xml-parsing/src/main/java/com/epam/training/exercise/ParserExercise.java

```diff
@@ -0,0 +1,61 @@
+package com.epam.training.exercise;
+
+import org.apache.batik.anim.dom.SVGDOMImplementation;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.List;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+public class ParserExercise {
+
+    public void parse() throws XMLStreamException, TransformerException, FileNotFoundException {
+        Document document = SVGDOMImplementation.getDOMImplementation().createDocument(SVGDOMImplementation.SVG_NAMESPACE_URI, "svg", null);
+
+        createDOMSource(document);
+        /**
+         * TODO
+         * 1. Parse XML -> készítsd el a saját koordináta kollekciódat az input.xml fájl alapján
+         * 2. Opcionális: Csoportosítsd államok/megyék szerint a városokat
+         *  Ha ezt a lépést kihagyod akkor az összes államot jelenítsd meg azonos színnel -> a szín legyen random generált
+         *  Ha elvégzed a csoportosítást akkor az államokat/megyéket külön random generált színnel jelenítsd meg (látszódjanak jól a határok!)
+         * 3.Végül koordináták jelenítsd meg az SVG fájlban egy pont formájában
+         */
+
+        TransformerFactory.newInstance().newTransformer().transform(new DOMSource(document), new StreamResult("OUTPUT SVG"));
+    }
+
+    private void createDOMSource(Document document) throws FileNotFoundException, XMLStreamException {
+        XMLStreamReader xmlStreamReader = XMLInputFactory.newFactory().createXMLStreamReader(new FileInputStream("INPUT XML"));
```

Comment:
I think parsing (i.e. creating the list of cities from *anything*, for example, an XML file) and producing output (i.e. writing data in *any* format, for example, into an SVG) are two rather distinct concerns/reasons for change, and we'd have more reusable/testable components if we could separate them. Not as if we'd ever want to reuse it, but I think it would be nice to show good (well, in many real-world scenarios, I guess) practices.

Of course, as I wrote in my other comment, we could leave it to the students, a discussion on why (and why not) one should write SOLID code would be beneficial as well.

Reply:
Yes, you are right. I have applied your suggestions. Thank you about it.